File: src/main/kotlin/com/ssabae/springdatajdbc/sample/support/AggregateReferenceValueExtractor.java
package com.ssabae.springdatajdbc.sample.support;

import org.springframework.data.jdbc.core.mapping.AggregateReference;

import javax.validation.valueextraction.ExtractedValue;
import javax.validation.valueextraction.ValueExtractor;

public class AggregateReferenceValueExtractor implements ValueExtractor<AggregateReference<?, @ExtractedValue ?>> {
    public AggregateReferenceValueExtractor() { }

    @Override
    public void extractValues(AggregateReference<?, ?> originalValue, ValueExtractor.ValueReceiver receiver) {
        receiver.value("id", originalValue.getId());
    }
}

File: src/main/kotlin/com/ssabae/springdatajdbc/sample/support/AggregateReferenceValueExtractor.kt
package com.ssabae.springdatajdbc.sample.support

import org.springframework.data.jdbc.core.mapping.AggregateReference
import javax.validation.valueextraction.ValueExtractor
import javax.validation.valueextraction.ValueExtractor.ValueReceiver

class AggregateReferenceValueExtractor : ValueExtractor<AggregateReference<*, *>> {
    override fun extractValues(
        originalValue: AggregateReference<*, *>,
        receiver: ValueReceiver
    ) {
        receiver.value("id", originalValue.id)
    }
}